compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -109,11 +109,59 @@ pub fn try_find_native_static_library(
     .break_value()
 }
 
+fn try_find_native_dynamic_library(sess: &Session, name: &str, verbatim: bool) -> Option<PathBuf> {
+    let formats = if verbatim {
+        vec![("".into(), "".into())]
+    } else {
+        // While the official naming convention for MSVC import libraries
+        // is foo.lib...
+        let os = (sess.target.staticlib_prefix.clone(), sess.target.staticlib_suffix.clone());
+        // ... Meson follows the libfoo.dll.a convention to
+        // disambiguate .a for static libraries
+        let meson = ("lib".into(), ".dll.a".into());
+        // and MinGW uses .a altogether
+        let mingw = ("lib".into(), ".a".into());
+        vec![os, meson, mingw]
+    };
+
+    let path = walk_native_lib_search_dirs(
+        sess,
+        LinkSelfContainedComponents::empty(),
+        None,
+        |dir, is_framework| {
+            if !is_framework {
+                for (prefix, suffix) in &formats {
+                    let test = dir.join(format!("{prefix}{name}{suffix}"));
+                    if test.exists() {
+                        return ControlFlow::Break(test);
+                    }
+                }
+            }
+            ControlFlow::Continue(())
+        },
+    )
+    .break_value();
+
+    match path {
```

Comment:
Could you mirror what `try_find_native_static_library` does here? (I.e. no fall back in this function.)

Reply:
This is not possible. The code path here also handles the CRT libraries, without fallback I get:

```
Building stage1 library artifacts {alloc, core, panic_abort, panic_unwind, proc_macro, std, sysroot, test, unwind} (x86_64-pc-windows-msvc)
   Compiling compiler_builtins v0.1.123
   Compiling core v0.0.0 (E:\rust\library\core)
   Compiling std v0.0.0 (E:\rust\library\std)
   Compiling windows-targets v0.0.0 (E:\rust\library\windows_targets)
   Compiling rustc-std-workspace-core v1.99.0 (E:\rust\library\rustc-std-workspace-core)
   Compiling alloc v0.0.0 (E:\rust\library\alloc)
   Compiling cfg-if v1.0.0
   Compiling rustc-demangle v0.1.24
   Compiling unwind v0.0.0 (E:\rust\library\unwind)
   Compiling rustc-std-workspace-alloc v1.99.0 (E:\rust\library\rustc-std-workspace-alloc)
   Compiling panic_abort v0.0.0 (E:\rust\library\panic_abort)
   Compiling panic_unwind v0.0.0 (E:\rust\library\panic_unwind)
   Compiling std_detect v0.1.5 (E:\rust\library\stdarch\crates\std_detect)
   Compiling hashbrown v0.14.5
error: could not find native static library `kernel32`, perhaps an -L flag is missing?
```